### ISSUE TITLE
Avoid recommending `context.log` for logs since it is actually emitting events

### ISSUE THREAD

**Issue by AndreaGiardini** [dagster-io/dagster#21207]
### Dagster version

1.7.1

### What's the issue?

Relevant slack discussion: https://discuss.dagster.io/t/16930496/hey-folks-wave-feedback-from-my-side-regarding-logging-and-e#d5317e86-39ec-4167-ba4e-c435cb3baa21

I have been working with dagster for quite some time and only now I realized that everything that goes into `context.log` is not actually a "log" but rather an "event". Consequentially that's not stored on S3/GCS but rather in the database.

I think this confusion has caused many people over time to fill up their DB much faster than they were expecting. If you look at Dagster discussions/issues there are many people asking how to remove runs because their DB fills up and I feel this could be one of the reasons. For instance, it's not unusual from some of our Data Scientist to pipe stderr/stdout into `context.log.info`, which caused our DB to grow massively over time.

The `Loggers` page in particular ( https://docs.dagster.io/concepts/logging/loggers ) is very misleading as it recommends piping your logs to `context.log`, which should be avoided if you don't want to end up with lots of records in your database.

### What did you expect to happen?

`context.log` should be used for logs, not events (maybe we should either rename this?)

### How to reproduce?

Log anything using `context.log` and the log file will end up in the database. I would expect log lines to end up in blob storage.

### Deployment type

Dagster Helm chart

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by AndreaGiardini** [comment]
@garethbrickman This should also be marked as "docs" rather than `feature-request`. The current recommendations for logging are inaccurate

**Comment by gofford** [comment]
To add to this I think there would be a lot of benefit to adding a documented example of how to properly write logs to stdout/stderr whilst also writing events to the event stream in the same process. This would allow users to write logs to storage and events to the database. It's not completely clear to me how this should be achieved with the dagster logging mechanisms at the moment.

**Comment by AndreaGiardini** [comment]
@garethbrickman I think this is still relevant: https://docs.dagster.io/guides/monitor/logging/python-logging#capturing-python-logs

**Comment by swertz** [comment]
I agree it would be good to clarify the documentation on this. In particular, I (and I assume, many other users in this situation) would like to use the standard python logging tools to write to the 'compute logs' visible in the stdout window, and stored on an S3, but I cannot seem to get it to work: the stdout window only shows what I `print()`, not what I log using python logging.

**Comment by ogmgaye** [comment]
> I agree it would be good to clarify the documentation on this. In particular, I (and I assume, many other users in this situation) would like to use the standard python logging tools to write to the 'compute logs' visible in the stdout window, and stored on an S3, but I cannot seem to get it to work: the stdout window only shows what I `print()`, not what I log using python logging.

Not sure if this'll help you because I run my instance in GCP but this is what I did.

First configure the log in dagster.yaml (or the cloud yaml, wherever you configure your instance). This will enable any python log to show up in the Dagster UI and to all go to `stdout` instead of `stderr`

```
python_logs:
   python_log_level: INFO
   managed_python_loggers: # This enables dagster to manage python logs I believe
       - python_logger
       - dagster
       - root
dagster_handler_config:
   handlers:
      json_stdout: # This overrides the default behavior of printing to stderr
          class: logging.StreamHandler
          formatter: json
          level: INFO
          stream: ext://sys.stdout
  formatters:
    json:
       '()': dagster.JsonLogFormatter # This outputs JSON/structured logs
```

then I used google-cloud-logging module to setup python logs to go to GCP logs with the correct severity. If AWS has something similar maybe it's just a matter of plugging that in.

Make sure this runs before any dagster code since python lazy loads the logging module otherwise if you try to log using the logging module you may see some strange behavior
``` 
import google.cloud.logging
import logging

# Instantiates a Cloud Logging client
client = google.cloud.logging.Client()

# Attaches the default Cloud Logging handler to the root logger.
# This captures logs at INFO level and higher by default.
client.setup_logging()

logging.getLogger("root").info("Root logger initialized...")
logging.getLogger("dagster").info("Dagster logger initialized...")
logging.getLogger("python_logger").info("Python logger initialized...")
```

test asset
```
import logging

@dg.asset
def log_context(context: OpExecutionContext): 
context.log.debug(f"Context log debug: Processing data from {sample_data_file}")
context.log.info(f"Context log info: Processing data from {processed_data_file}")
context.log.warning(f"Context log warning: Processing data from {processed_data_file}")
context.log.error(f"Context log error: Processing data from {processed_data_file}")
context.log.critical(f"Context log critical: Processing data from {processed_data_file}")

my_logger = logging.getLogger('python_logger')
my_logger.info(context.log.name)
my_logger.info(f"Context log dict: {context.log.__dict__}")
my_logger.info(f"Context dict: {context.run_id}", extra={"run_id": context.run_id})
my_logger.debug(f"Python logger debug: Processing data from {sample_data_file}")
my_logger.info(f"Python logger info: Processing data from {processed_data_file}")
my_logger.warning(f"Python logger warning: Processing data from {processed_data_file}")
my_logger.error(f"Python logger error: Processing data from {processed_data_file}")
my_logger.critical(f"Python logger critical: Processing data from {processed_data_file}")
my_logger.info(
msg=f"Python logger log JSON log: Processing data from {processed_data_file}",
extra={"json_fields": {
"pipeline_name": context.job_name,
"run_id": context.run_id,
}}
)

return "Log context" 
```

<img width="1737" height="574" alt="Image" src="https://github.com/user-attachments/assets/308d64da-9a44-4cf3-a358-6448f92bc2dd" />
<img width="1736" height="589" alt="Image" src="https://github.com/user-attachments/assets/1b9d8b59-270f-4989-a1a9-473aa8ec3c0f" />

**Comment by swertz** [comment]
Thanks @ogmgaye ! But does this also populate the events table in the dagster database? From your screenshots, it seems that the logs sent to `my_logger` also result in "events" which end up in the dagster database. This can cause it to fill up very quickly (and can also slow down the pipelines significantly, since a DB transaction is made for every log message!).

**Comment by ogmgaye** [comment]
@swertz I'm not too sure but what you've stated makes sense. But does Dagster not emit events for everything already? Only difference now would be making sure when GCP reads these event logs, it categorizes it appropriately? 

I'm ignorant in this space so I'm not too sure. Primarily worked on this to help my DE team get better logs.
